src/api/firebase.js

```diff
@@ -87,10 +91,50 @@ export async function updateItem(listId, itemId, checked) {
 	const listItemRef = doc(db, listId, itemId);
 	const listItemSnap = await getDoc(listItemRef);
 
-	let totalPurchases = listItemSnap.data().totalPurchases;
+	// Retrieve item property values stored in Firebase
+	let { totalPurchases, dateLastPurchased, dateCreated, dateNextPurchased } =
+		listItemSnap.data();
+
+	//This function returns the next purchase date in milliseconds
+	const getNextPurchaseDate = () => {
+		// Function estimating previous purchase interval
+		const previousEstimate = dateLastPurchased
+			? getDaysBetweenDates(dateNextPurchased, dateLastPurchased)
+			: getDaysBetweenDates(dateNextPurchased, dateCreated);
+
+		// This declares the number of days since the last transaction based on dateLastPurchased being a null or existing value
+		const numDaysSinceLastTransaction = dateLastPurchased
+			? // If dateLastPurchased returns TRUE, we get number of days between *dateLastPurchased* and current date
+			  getDaysBetweenDates(new Date(), dateLastPurchased.toDate())
+			: // If dateLastPurchased returns FALSE, we get number of days between *dateCreated* and current date
+			  getDaysBetweenDates(new Date(), dateCreated.toDate());
+
+		// The estimated next purchase date is calculated by adding two values:
+		//	- The current date in milliseconds
+		//	- The estimated number of days until the next purchase converted into milliseconds
+		const nextPurchaseDate = new Date(
+			new Date().getTime() +
+				calculateEstimate(
+					previousEstimate,
+					numDaysSinceLastTransaction,
+					totalPurchases,
+				) *
+					ONE_DAY_IN_MILLISECONDS,
+		);
 
```

Comment:
I think as a former English teacher, this mountain of logic counts as an A+ essay.  Phew y'all did a ton of work and thinking to make this work.  Impressive and clear.